src/ibtracert.c

```diff
@@ -811,9 +826,36 @@ int main(int argc, char **argv)
 	argc -= optind;
 	argv += optind;
 
-	if (argc < 2)
+	if (argc < 2 && ports_file == NULL) 
 		ibdiag_show_usage();
 
+	if (argc == 2) {
+		srcid = strdup(argv[0]);
+		if (srcid == 0)
+			IBEXIT("out of memory, strdup for srcid failed");
+		dstid = strdup(argv[1]);
+		if (dstid == 0)
+			IBEXIT("out of memory, strdup for dstid failed");
+     	}
+
+	if (ports_file != NULL) {
+		ports_fd = fopen(ports_file, "r");
+		if (!ports_fd)
+			IBEXIT("cannot open ports-file %s", ports_file);
+        	if (!fgets(ports_buffer, sizeof(ports_buffer), ports_fd)) 
+			IBEXIT("read of ports-file, %s, failed", ports_file);
+		srcid = (char*) malloc(20*sizeof(char));
```

Comment:
Where is this memory freed?  Don't you need to store an array of srcid's and dstid's?  Ok never mind I see it now...  Just have 2 static buffers.

char srcbuf[20];
char dstbuf[20];

And if we are reading the file:

srcid = srcbuf;
dstid = dstbuf;

And why are we only reading the first line of the file?

Can the file contain comments?  ie.

     # this is a ports pair file
     # and some more indented comments

Reply:
The whole file gets read, see lines 919-935.
Will add the ability to include comments.